File: back-end/src/main/java/vn/bizup/crete/models/entities/Registration.java
package vn.bizup.crete.models.entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Registration {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer registrationId;
    private Integer patientId;
    private String patientIdentity;
    private String patientName;
//    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date createDate;
    private Double totalAmount;
    private boolean deleted;
    private boolean closed;
}

File: front-end/src/app/modules/examination/models/registration.ts
export interface Registration {
  registrationId?: number;
  createDate?: Date;
  patientId?: number;
  patientIdentity?: string;
  patientName?: string;
  totalAmount?: number;
  services?: RegistrationDetail[];
}

export interface RegistrationDetail {
  registrationDetailId?: number;
  registrationId?: number;
  serviceId?: number;
  serviceName?: string;
  price?: number;
}
